rfcs/text/0000-application-monitoring-for-ga.md

```diff
@@ -0,0 +1,146 @@
+ 
+ Your monitoring coverage should be as "[simple as possible, no simpler](https://landing.google.com/sre/sre-book/chapters/monitoring-distributed-systems/#as-simple-as-possible-no-simpler-lqskHx)" to diagnose common failure modes of your system. Some common examples are included below.
+ 
+ * Common failure modes of Elasticsearch clusters are related to JVM memory pressure and minimum shard storage space. These are good things to have on a dashboard.
+ * HTTP APIs should have a dashboard summarizing rates of requests by HTTP status class (`2xx`, `3xx`, `4xx`, `5xx`) and request latencies by percentile (commonly 50th, 75th, 95th, 99th). Adding an optional route and method dimensions to each metric will help you pinpoint what is failing when errors occur.
+ * Queues should be monitored for length, size (bytes), consumption rates by status, and age of oldest message.
+ 
+### Alerts
+
+We need alerts (or alarms) in order to be notified of problems in the system.
+
+Configure alerts in the [dcp-monitoring](https://github.com/HumanCellAtlas/dcp-monitoring) repo with a [CloudWatch Alarm](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#CloudWatchAlarms). All black-box Route 53 health checks should be configured with an alert. Optionally, you may choose to set thresholds on some of your white-box monitoring metrics if they are highly-correlated with system failure (e.g. an EC2 node running out of disk space).
+
+Alerts are currently propagated to the Slack channel specified in the CloudWatch Alarm configuration.
+
+### Dashboards
+
+Dashboards answer basic, frequently-asked questions about your service. Ideally, they should be a high-level, easy-to-read summary of the metrics from a system's white-box monitoring.
+
+Create and host dashboards for your service using our [Grafana deployments](https://github.com/HumanCellAtlas/metrics). You can template your dashboards to generalize to multiple deployments of your software using [dcp-monitoring](https://github.com/HumanCellAtlas/dcp-monitoring).
+
+![](../images/0000-dashboard.png)
+*An example dashboard visualizing white-box metrics for an HTTP API backed by an Elasticsearch cluster to store state. The performance of the HTTP API is monitored in the first row with HTTP request rates, statuses, and latencies. The second row monitors the metrics that would predict system failure due to Elasticsearch cluster overload.*
+
+### Status and availability monitoring
+
+System status and availability monitoring is provided by a [Status API](https://github.com/HumanCellAtlas/status-api), which then can be summarized on a [status page](https://humancellatlas.github.io/).
+
+### Leadership
+
+The future DevSecOps group at the Broad Institute will lead program management for the monitoring effort.
```

Comment:
Please could you elaborate. Who will make up this group? What does program management mean (project mananagement?). What is the roadmap for establishing this group? What is the governance structure?

Reply:
This is outside the scope of this document. Happy to follow up on this in a different channel.

Reply:
Would be great to know more. Also in that case, does this consideration need to be in the RFC?

Reply:
There were some questions earlier about how this would work, so I added this very brief bit. Debatable, but since those questions did come up, I'll leave it.